### ISSUE TITLE
use of np.fmax with 3 arrays in equations.variability prob

### ISSUE THREAD

**Issue by dgketchum** [mapbox/rio-cloudmask#15]
Hi @perrygeo,
I love all the work you've done and put on github, it has taught me a lot and made my life of working with landsat data easier!

I'm putting your implementation of fmask equations in a class and noticed a possible misuse of numpy.fmax:

```
def variability_prob(ndvi, ndsi, whiteness):
 return 1.0 - np.fmax(np.absolute(ndvi), np.absolute(ndsi), whiteness)
```
where fmax should only evaluate two arrays at a time, e.g. in the following code indices in which c is greatest are ignored:

```
>>> a
array([[177,  44,  99],
       [198, 108, 206],
       [180,  10,  35]])
>>> b
array([[101,  64,  53],
       [  5, 169,  82],
       [206, 111, 252]])
>>> c
array([[149,  87,  36],
       [210, 111,  83],
       [189, 208, 106]])
>>> np.fmax(a, b, c)
array([[177,  64,  99],
       [198, 169, 206],
       [206, 111, 252]])
```
This is fixed with 

```
        ndi_max = np.fmax(np.absolute(self.ndvi), np.absolute(self.ndsi))
        f_max =  1.0 - np.fmax(ndi_max, whiteness)
        return f_max
```

I'm planning on continued work with fmask.  If you are open to PRs, I might work on equations.py (and tests) and seperate the functions from my fmask class.

Cheers

**Comment by perrygeo** [comment]
@dgketchum Thanks for catching this bug!  Not only is the max being calculated incorrectly, It seems like the third argument to `numpy.fmax` is an [optional output array](https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.fmax.html) which means the code is mutating the `whiteness` array as well.

I would gladly accept a PR with the fix above. Looks like the correct way. Thanks for digging in! 👍

**Comment by perrygeo** [comment]
resolved by #16
